controller/sync.go

```diff
@@ -617,15 +625,29 @@ func (sc *syncContext) terminate() {
 }
 
 func (sc *syncContext) deleteResource(task *syncTask) error {
-	sc.log.WithFields(log.Fields{"task": task}).Debug("deleting task")
+	sc.log.WithFields(log.Fields{"task": task}).Debug("deleting resource")
 	apiResource, err := kube.ServerResourceForGroupVersionKind(sc.disco, task.groupVersionKind())
 	if err != nil {
 		return err
 	}
 	resource := kube.ToGroupVersionResource(task.groupVersionKind().GroupVersion().String(), apiResource)
 	resIf := kube.ToResourceInterface(sc.dynamicIf, apiResource, resource, task.namespace())
 	propagationPolicy := metav1.DeletePropagationForeground
-	return resIf.Delete(task.name(), &metav1.DeleteOptions{PropagationPolicy: &propagationPolicy})
+	err = resIf.Delete(task.name(), &metav1.DeleteOptions{PropagationPolicy: &propagationPolicy})
+	if err != nil {
+		return err
+	}
+	// double-check the delete was successful, and fail the sync if not
```

Comment:
I wanted to fail this rather than wait because a job/workflow should have completed. If not, perhaps a manual intervention is the best thing to do.